sections/3-architecture.md

```diff
@@ -366,3 +366,104 @@ primitives (e.g. ints, strings, etc.), so usage of named
 parameters makes the code more resistant to change and less error-prone,
 versus relying on positioning. The style of indentation chosen here makes
 the instantiation look like a `Map` or a JSON object if you want.
+
+### 3.6 SHOULD put type-class instances as close to the call site as possible
+
+Type-classes are a powerful feature close to functional programming mindset. It 
+favours composition and is central to core `Scala` libraries like 
+[`circe`](https://circe.github.io/circe/), [`cats`](https://typelevel.org/cats/) 
+or even [`scalacheck`](https://www.scalacheck.org/).
+
+One may keep wondering where to put the type-class instances. Each type-class
+has a purpose. Some may be more general like purely functional abstractions 
+from `cats`, some more specific like *json* serialization from `circe`. A good 
+practice is to *put them as close to the call site as possible*. For example:
+
+```scala
+// /core/model/ProductDetail.scala
+
+final case class ProductDetail(id: Int, label: String, description: String)
+
+object ProductDetail {
+  implicit val productDetailEncoder= deriveEncoder
+}
```

Comment:
Similar here. It's kind of not clear when this implicit it put into scope.

Reply:
That is issue sort of here. During when resolving implicit for certain data type, the compiler looks at its companion object. Problem is that call-site specific instances (such as json serialization) do not belong there.